### PR TITLE
Move PixelBarrelName and PixelEndcapName into TrackerCommon to avoid circular dependency

### PR DESCRIPTION
**Pull request by davidlange6**

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by perrotta** [comment]
+1
- Files moved as stated
- Jenkins tests pass

**Comment by Martin-Grunewald** [comment]
+1

File: DataFormats/TrackerCommon/interface/PixelPannelType.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #ifndef DataFormats_SiPixelDetId_PixelPannelType_H
```
Comment (perrotta) [inline DataFormats/TrackerCommon/interface/PixelPannelType.h:1]: Just noticing that in the other moved headers you also updated the header guard....

**Comment by VinInn** [comment]
Would not be easier to reduce the number of packages (and libs!)?
I do not see any reason to have SiPixelDetId, SiStripDetId etc in their own package

**Comment by davidlange6** [comment]
Its all just a sed command. I made minimal changes to solve a technical problem.

On Sep 8, 2020, at 9:30 AM, Vincenzo Innocente <notifications@github.com<mailto:notifications@github.com>> wrote:



Would not be easier to reduce the number of packages (and libs!)?
I do not see any reason to have SiPixelDetId, SiStripDetId etc in their own package

**Comment by VinInn** [comment]
ok, maybe TRK-DPG should have a look and take action

**Comment by davidlange6** [comment]
please test

**Comment by perrotta** [comment]
+1
- Files moved as stated
- Jenkins tests pass

**Comment by tlampen** [comment]
+alca

**Comment by mmusich** [comment]
> ok, maybe TRK-DPG should have a look and take action

@VinInn please open a CMSSW issue, so it doesn't fall of the radar (given the rather low priority)

**Comment by VinInn** [comment]
I'll leave it to reco conveners (if they think is worth/relevant/correct)

**Comment by mmusich** [comment]
> I'll leave it to reco conveners (if they think is worth/relevant/correct)

As far as I can tell, the packages involved `SiPixelDetId` and `SiStripDetId` are rather under `simulation` signature:

https://github.com/cms-sw/cms-bot/blob/master/categories_map.py#L1425 

https://github.com/cms-sw/cms-bot/blob/master/categories_map.py#L1428

**Comment by Martin-Grunewald** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
@cms-sw/db-l2 could you please have a look and sign, if you think so? It is just a quite technical PR, and since it involves several files, it is proabably worth trying to merge it now and avoid thus possible conflicts in future...

**Comment by ggovi** [comment]
+1

**Pull request merged**